collector-http/src/main/java/zipkin2/collector/otel/http/SpanTranslator.java

```diff
@@ -46,14 +49,24 @@ final class SpanTranslator {
 
   static final String ERROR_TAG = "error";
 
-  static List<zipkin2.Span> translate(ExportTraceServiceRequest otelSpans) {
+  SpanTranslator(CollectorMetrics metrics) {
+    this.metrics = metrics;
+  }
+
+  List<zipkin2.Span> translate(ExportTraceServiceRequest otelSpans) {
     List<zipkin2.Span> spans = new ArrayList<>();
     List<ResourceSpans> spansList = otelSpans.getResourceSpansList();
     for (ResourceSpans resourceSpans : spansList) {
       for (ScopeSpans scopeSpans : resourceSpans.getScopeSpansList()) {
         InstrumentationScope scope = scopeSpans.getScope();
         for (io.opentelemetry.proto.trace.v1.Span span : scopeSpans.getSpansList()) {
-          spans.add(generateSpan(span, scope, resourceSpans.getResource()));
+          try {
+            spans.add(generateSpan(span, scope, resourceSpans.getResource()));
+          }
+          catch (RuntimeException e) {
```

Comment:
personally i would capture at call site calling translator if you can, as this is more like any other issue about malformed inclusing bad trace ID but not limited to that

Reply:
If it is caught outside of the translator call, the valid spans in the trace data may also be lost. Is that expected?

Reply:
I think so, as consider if it were json and one span was in incorrect format, you would lose the message. I think bugs should be fixed basically

Reply:
say I'm wrong, we get an issue asking for a different behavior also. meanwhile there isn't code at possibly not the right abstraction ;)

Reply:
The IOException that occurs when converting a protobuf payload to a Java object occurs before calling translate and is handled. This runtime exception occurs when converting an otel span to a zipkin span, and may be an illegal argument or null pointer exception. If a span with an invalid value is inserted among multiple valid spans, is it okay to drop all of them? This is a rare case, so I don't really care.

Reply:
probably one thing that I didn't highlight is that when in doubt we can refer to the actual collector.java which handles exception and rte the same way https://github.com/openzipkin/zipkin/blob/master/zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java#L171-L180

Reply:
basically what I mean is that error handling and stats handling were all done in collector.java. it is true that we cannot always re-use that type, but the behavior expectations we can borrow from it anyway

Reply:
Yes, I looked at the implementation of Collector and left incrementing up to Collector as much as possible. I thought I would accept valid spans as much as possible. However, if it's not a good design for Translator to depend on CollectorMetrics, I'll count Dropped outside of Translator.

Reply:
consider my opinion non binding basically i feel this code is a little cleaner not injecting collectormetrics to it. i will approve either way. your call

Reply:
> basically i feel this code is a little cleaner not injecting collectormetrics to it.

Agreed.

I considered passing a callback to decouple, but that would be too much for some rare cases, so I simply caught the exception outside the translator.